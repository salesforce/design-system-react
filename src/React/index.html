<!doctype html public "library">
<meta charset=utf-8/>
<head>
	<title>Selectlist</title>
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
</head>
<body>
<script src="/dist/react/build.js"></script>
</body>
</html>