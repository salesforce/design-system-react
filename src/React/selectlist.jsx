//test

var React = require('react');
import {SelectlistCore} from "../SelectlistCore";
var DropdownButton = require('react-bootstrap/lib/DropdownButton');
var MenuItem = require('react-bootstrap/lib/MenuItem');

var DATA = {
	selected: 'None selected',
	id: 'selectlist1',
	items: [
		{ id: 0, name: 'tacos', type: 'mexican' },
		{ id: 1, name: 'burrito', type: 'mexican' },
		{ id: 2, name: 'tostada', type: 'mexican' },
		{ id: 3, name: 'hush puppies', type: 'southern' }
	]
};

var menuItemSelected = (eventKey, href, target)=> {
	DATA.selected = DATA.items[eventKey].name;
};
//Selectlist.handleMenuItemClicked
var menuItems = DATA.items.map( (menuItem) => {
	return <MenuItem eventKey={menuItem.id}>{menuItem.name}</MenuItem>;
});

var Selectlist = React.createClass({
	getInitialState: function() {
		return {
			selectedKey: undefined
		};
	},

	handleMenuItemClicked: function(eventKey, href, target) {
		this.setState({
			selectedKey: eventKey
		});
	},

	menuItems: function() {
		return DATA.items.map( (menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>{menuItem.name}</MenuItem>;
		});
	},

	selected: function() {
		return typeof(this.state.selectedKey) === 'undefined' ? 'None Selected' : DATA.items[this.state.selectedKey].name;
	},

	render () {
		return (
			<div>
				<h1>SelectList</h1>
				<ul>{this.menuItems}</ul>
				<DropdownButton title={this.selected()} key={DATA.id}>{this.menuItems()}</DropdownButton>
			</div>
		);
	}
});

React.render(<Selectlist/>, document.body, () => {});