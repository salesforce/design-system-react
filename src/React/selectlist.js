// SELECTLIST CONTROL

// core
import {Landmark} from '../Landmark';
import {SelectlistCore} from '../SelectlistCore';

// framework specific
import React from 'react';
import DropdownButton from 'react-bootstrap/lib/DropdownButton';
import MenuItem from 'react-bootstrap/lib/MenuItem';

export var Selectlist = React.createClass(Object.assign({}, SelectlistCore, {	
	propTypes: {
		disabled: React.PropTypes.bool,
		selection: React.PropTypes.number,
		collection: React.PropTypes.array
	},
	
	getInitialState () {
		return this.__getInitialState();
	},
		
	menuItems () {
		return this.props.collection.map((menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>
					{menuItem.name}
				</MenuItem>;
		});
	},
	
	componentWillReceiveProps(nextProps) {
		this.__initializeOptions(nextProps);
	},
	
	render () {
		var selection = this.getSelection();
		
		return (
			<DropdownButton className={this.cssClasses.CONTROL} disabled={this.state.disabled} title={selection ? selection.name : 'None selected'} key={this.props.id}>{this.menuItems()}</DropdownButton>
		);
	},
	
	componentWillMount () {
		this.elements = {
			wrapper: {
				toggleClass: function (cssClass, state) {
					Landmark.log(cssClass, state);
				}
			}
		}
		
		this.__constructor(this.props);
	},
	
	handleMenuItemClicked (eventKey, href, target) {
		this.setSelectionByKey('id', eventKey);
	}
}));