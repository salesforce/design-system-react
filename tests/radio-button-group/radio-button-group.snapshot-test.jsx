/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { renderMarkup } from '../snapshot-helpers';

import SnapshotExample from '../../examples/radio-button-group/base';

test('Radio Button Group Base DOM Snapshot', () => {
	const domTree = renderer.create(
		<SnapshotExample name="dayOfWeek" errorId="dayOfWeekError" />,
	).toJSON();
	expect(domTree).toMatchSnapshot();
});

test('Radio Button Group Base HTML Snapshot', () => {
	expect(renderMarkup(SnapshotExample, { name: 'dayOfWeek', errorId: 'dayOfWeekError' })).toMatchSnapshot();
});

test('Radio Button Group Disabled DOM Snapshot', () => {
	const domTree = renderer.create(
		<SnapshotExample name="dayOfWeek" errorId="dayOfWeekError" disabled />,
	).toJSON();
	expect(domTree).toMatchSnapshot();
});

test('Radio Button Group Required DOM Snapshot', () => {
	const domTree = renderer.create(
		<SnapshotExample name="dayOfWeek" errorId="dayOfWeekError" required />,
	).toJSON();
	expect(domTree).toMatchSnapshot();
});
