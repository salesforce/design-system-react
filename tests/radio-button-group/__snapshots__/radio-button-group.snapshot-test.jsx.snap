// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Button Group Base DOM Snapshot 1`] = `
<div>
  <fieldset
    className="slds-form-element"
  >
    <legend
      className="slds-form-element__legend slds-form-element__label"
    >
      Day of week
    </legend>
    <div
      className="slds-form-element__control"
    >
      <div
        className="slds-radio_button-group"
      >
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={false}
            disabled={undefined}
            id="Mon"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Mon"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Mon"
          >
            <span
              className="slds-radio_faux"
            >
              Mon
            </span>
          </label>
        </span>
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={true}
            disabled={undefined}
            id="Tue"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Tue"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Tue"
          >
            <span
              className="slds-radio_faux"
            >
              Tue
            </span>
          </label>
        </span>
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={false}
            disabled={undefined}
            id="Wed"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Wed"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Wed"
          >
            <span
              className="slds-radio_faux"
            >
              Wed
            </span>
          </label>
        </span>
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={false}
            disabled={undefined}
            id="Thu"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Thu"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Thu"
          >
            <span
              className="slds-radio_faux"
            >
              Thu
            </span>
          </label>
        </span>
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={false}
            disabled={undefined}
            id="Fri"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Fri"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Fri"
          >
            <span
              className="slds-radio_faux"
            >
              Fri
            </span>
          </label>
        </span>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Radio Button Group Base HTML Snapshot 1`] = `
"<div>
  <fieldset class=\\"slds-form-element\\">
    <legend class=\\"slds-form-element__legend slds-form-element__label\\">Day of week</legend>
    <div class=\\"slds-form-element__control\\">
      <div class=\\"slds-radio_button-group\\"><span class=\\"slds-button slds-radio_button\\"><input type=\\"radio\\" id=\\"Mon\\" name=\\"dayOfWeek\\" value=\\"Mon\\" aria-describedby=\\"dayOfWeekError\\"/><label class=\\"slds-radio_button__label\\" for=\\"Mon\\"><span class=\\"slds-radio_faux\\">Mon</span></label>
        </span><span class=\\"slds-button slds-radio_button\\"><input type=\\"radio\\" id=\\"Tue\\" name=\\"dayOfWeek\\" value=\\"Tue\\" checked=\\"\\" aria-describedby=\\"dayOfWeekError\\"/><label class=\\"slds-radio_button__label\\" for=\\"Tue\\"><span class=\\"slds-radio_faux\\">Tue</span></label>
        </span><span class=\\"slds-button slds-radio_button\\"><input type=\\"radio\\" id=\\"Wed\\" name=\\"dayOfWeek\\" value=\\"Wed\\" aria-describedby=\\"dayOfWeekError\\"/><label class=\\"slds-radio_button__label\\" for=\\"Wed\\"><span class=\\"slds-radio_faux\\">Wed</span></label>
        </span><span class=\\"slds-button slds-radio_button\\"><input type=\\"radio\\" id=\\"Thu\\" name=\\"dayOfWeek\\" value=\\"Thu\\" aria-describedby=\\"dayOfWeekError\\"/><label class=\\"slds-radio_button__label\\" for=\\"Thu\\"><span class=\\"slds-radio_faux\\">Thu</span></label>
        </span><span class=\\"slds-button slds-radio_button\\"><input type=\\"radio\\" id=\\"Fri\\" name=\\"dayOfWeek\\" value=\\"Fri\\" aria-describedby=\\"dayOfWeekError\\"/><label class=\\"slds-radio_button__label\\" for=\\"Fri\\"><span class=\\"slds-radio_faux\\">Fri</span></label>
        </span>
      </div>
    </div>
  </fieldset>
</div>"
`;

exports[`Radio Button Group Disabled DOM Snapshot 1`] = `
<div>
  <fieldset
    className="slds-form-element"
  >
    <legend
      className="slds-form-element__legend slds-form-element__label"
    >
      Day of week
    </legend>
    <div
      className="slds-form-element__control"
    >
      <div
        className="slds-radio_button-group"
      >
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={false}
            disabled={true}
            id="Mon"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Mon"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Mon"
          >
            <span
              className="slds-radio_faux"
            >
              Mon
            </span>
          </label>
        </span>
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={false}
            disabled={true}
            id="Tue"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Tue"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Tue"
          >
            <span
              className="slds-radio_faux"
            >
              Tue
            </span>
          </label>
        </span>
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={false}
            disabled={true}
            id="Wed"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Wed"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Wed"
          >
            <span
              className="slds-radio_faux"
            >
              Wed
            </span>
          </label>
        </span>
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={false}
            disabled={true}
            id="Thu"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Thu"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Thu"
          >
            <span
              className="slds-radio_faux"
            >
              Thu
            </span>
          </label>
        </span>
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={false}
            disabled={true}
            id="Fri"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Fri"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Fri"
          >
            <span
              className="slds-radio_faux"
            >
              Fri
            </span>
          </label>
        </span>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Radio Button Group Required DOM Snapshot 1`] = `
<div>
  <fieldset
    className="slds-form-element"
  >
    <legend
      className="slds-form-element__legend slds-form-element__label"
    >
      <abbr
        className="slds-required"
        title="required"
      >
        *
      </abbr>
      Day of week
    </legend>
    <div
      className="slds-form-element__control"
    >
      <div
        className="slds-radio_button-group"
      >
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={false}
            disabled={undefined}
            id="Mon"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Mon"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Mon"
          >
            <span
              className="slds-radio_faux"
            >
              Mon
            </span>
          </label>
        </span>
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={true}
            disabled={undefined}
            id="Tue"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Tue"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Tue"
          >
            <span
              className="slds-radio_faux"
            >
              Tue
            </span>
          </label>
        </span>
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={false}
            disabled={undefined}
            id="Wed"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Wed"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Wed"
          >
            <span
              className="slds-radio_faux"
            >
              Wed
            </span>
          </label>
        </span>
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={false}
            disabled={undefined}
            id="Thu"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Thu"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Thu"
          >
            <span
              className="slds-radio_faux"
            >
              Thu
            </span>
          </label>
        </span>
        <span
          className="slds-button slds-radio_button"
        >
          <input
            aria-describedby="dayOfWeekError"
            checked={false}
            disabled={undefined}
            id="Fri"
            name="dayOfWeek"
            onChange={[Function]}
            type="radio"
            value="Fri"
          />
          <label
            className="slds-radio_button__label"
            htmlFor="Fri"
          >
            <span
              className="slds-radio_faux"
            >
              Fri
            </span>
          </label>
        </span>
      </div>
    </div>
  </fieldset>
</div>
`;
